### ISSUE TITLE
[P1] Elementwise CTM convergence certifies NaN environments with zero residual

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#974]
> 🤖 **AI-generated issue** - written by Codex from the requested iPEPS code review.

## Finding

The square-lattice elementwise convergence check silently ignores NaN leaf differences. It can return `converged=True, sv_diff=0.0` while every returned environment tensor is non-finite. This affects the shared forward convergence loop used by the Python-loop and implicit-AD paths; `CTMConfig.ctm_conv_method` defaults to `"elementwise"`.

Verified on CPU with JAX x64 at `4090f98be5c3efdce20a28663536f926171f857b`. This reproduction deliberately injects a corrupt warm-start environment to test the failure-reporting contract, not the numerical stability of a particular physical state.

## Reproduction

Run with `JAX_PLATFORMS=cpu JAX_ENABLE_X64=1 uv run python`:

```python
import jax
import jax.numpy as jnp
from tenax.algorithms.ipeps import _wrap_as_dense_tensor
from tenax.algorithms._ctm_tensor_init import initialize_ctm_tensor_env
from tenax.algorithms._ctm_tensor_convergence import (
    SINGLE_SITE_NEIGHBORS, _max_env_leaf_diff,
)
from tenax.algorithms._ctm_python_loop import python_loop_ctm_converge

A = _wrap_as_dense_tensor(jnp.ones((1, 1, 1, 1, 2)))
env = initialize_ctm_tensor_env(A, 1)
bad = jax.tree.map(lambda x: jnp.full_like(x, jnp.nan), env)
print("helper:", _max_env_leaf_diff(env, bad))

for method in ("elementwise", "sv"):
    envs, info = python_loop_ctm_converge(
        {(0, 0): A}, SINGLE_SITE_NEIGHBORS,
        chi=1, max_iter=4, min_iter=2, conv_method=method,
        env_init={(0, 0): bad}, renormalize=True,
    )
    finite = all(
        bool(jnp.all(jnp.isfinite(x))) for x in jax.tree.leaves(envs)
    )
    print(method, info.converged, info.iterations, info.sv_diff, finite)
```

Observed:
```text
helper: 0.0
elementwise True 2 0.0 False
sv False 4 inf False
```

Expected: a non-finite environment must never be certified as converged. Return a non-finite/failure residual and `converged=False`, or raise a clear numerical error.

## Root Cause

- [`_max_env_leaf_diff`](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/_ctm_tensor_convergence.py#L813-L821) uses Python `max(max_diff, diff)`. `max(0.0, float("nan"))` is `0.0`, so each NaN difference disappears.
- [The loop](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/_ctm_loop_core.py#L255-L271) sees that zero and returns success before its finite-metric plateau guard can help. The nearby comment assumes a NaN leaf makes the aggregate non-finite, but the reducer prevents that.
- [Honeycomb convergence](https://github.com/tenax-lab/tenax/blob/4090f98be5c3efdce20a28663536f926171f857b/src/tenax/algorithms/_ctm_honeycomb_convergence.py#L25-L68) has the same defect in both elementwise and SVD reducers: `if diff > worst` ignores NaN. Independently reproduced by taking the finite environments from `initialize_honeycomb_env`, mapping every leaf to NaN, and calling `check_honeycomb_convergence(old, bad, method=..., tol=1e-10)`: both methods return `True`.

## Suggested Fix and Coverage

1. Reject/non-finitely mark each leaf difference before aggregating it; do not rely on Python comparison semantics to propagate NaN.
2. Keep non-finite sweeps ineligible for convergence and plateau success. Audit the honeycomb reducers in the same change.
3. Add finite-to-NaN, NaN-to-NaN, single-corrupt-leaf, and infinity cases, plus a loop-level assertion that the returned metadata cannot certify a corrupted environment.
4. Preserve the healthy finite and square-lattice SV behavior.

This is distinct from #898's rank-one spectral blindness: the spectral guard rejects this reproduction, but the elementwise reducer still accepts it. Focused baseline: 46 tests passed across CTM loop, metadata, honeycomb convergence, and chi-bump suites; these non-finite cases are not covered.
